activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/JobExecutorIT.java

```diff
@@ -0,0 +1,387 @@
+                                  .setCurrentTime(new Date(startTime.getTime() + ((60 * 60 * 1000) + 5000)));
+
+        // then
+        await("the process should start and no more timer jobs should exist")
+           .untilAsserted(() -> {
+               assertThat(runtimeService.createProcessInstanceQuery()
+                                        .processDefinitionId(processDefinitionId)
+                                        .count()).isEqualTo(1);
+               
+               assertThat(managementService.createTimerJobQuery()
+                                           .processDefinitionId(processDefinitionId)
+                                           .count()).isEqualTo(0);
+           });
+
+        // timer event has been fired
+        assertThat(timerFired.await(1, TimeUnit.MINUTES)).as("should fire timer")
+                                                         .isTrue();
+
+        // job event has been completed
+        assertThat(jobCompleted.await(1, TimeUnit.MINUTES)).as("should complete job")
+                                                            .isTrue();
+        // message handler is invoked
+        verify(jobMessageHandler).handleMessage(ArgumentMatchers.<Message<?>>any());
+    }
+    
+    @SuppressWarnings("deprecation")
+    @Test
+    public void testBoundaryTimerEvent() throws Exception {
+        CountDownLatch jobsCompleted = new CountDownLatch(1);
+        CountDownLatch timerScheduled = new CountDownLatch(1);
+        CountDownLatch timerFired = new CountDownLatch(1);
+
+        // Set the clock fixed
+        Date startTime = new Date();
+
+        runtimeService.addEventListener(new CountDownLatchActvitiEventListener(timerScheduled),
+                                        ActivitiEventType.TIMER_SCHEDULED);
+
+        runtimeService.addEventListener(new CountDownLatchActvitiEventListener(timerFired),
+                                        ActivitiEventType.TIMER_FIRED);
+
+        runtimeService.addEventListener(new CountDownLatchActvitiEventListener(jobsCompleted),
+                                        ActivitiEventType.JOB_EXECUTION_SUCCESS);
+
+        // when
+        ProcessInstance pi = runtimeService.startProcessInstanceByKey(TEST_BOUNDARY_TIMER_EVENT);
+
+        // then
+        assertThat(pi).isNotNull();
+        
+        await("the timer job should be created")
+            .untilAsserted(() -> {
+                assertThat(managementService.createTimerJobQuery()
+                                            .processInstanceId(pi.getId())
+                                            .count()).isEqualTo(1);
+            });
+
+        // After setting the clock to time '5 minutes and 5 seconds', the timer should fire
+        processEngineConfiguration.getClock().setCurrentTime(new Date(startTime.getTime() + ((5 * 60 * 1000) + 5000)));
+
+        // timer event has been scheduled
+        assertThat(timerScheduled.await(1, TimeUnit.MINUTES)).as("should schedule timer")
```

Comment:
Sounds a really long wait.

Reply:
It should proceed as soon as latch is triggered, so it is not really waiting for 1 minute